### ISSUE TITLE
okta_policy_rule_signon SPECIFIC_IDP Quit Working With Provider Version 3.25.1

### ISSUE THREAD

**Issue by kellycampbe** [okta/terraform-provider-okta#1126]
Starting with provider version 3.25.1 when creating _okta_policy_rule_signon_ with _identity_provider  = "SPECIFIC_IDP"_ it is no longer selecting the IdP.

Below is example Terraform.

```
resource "okta_policy_signon" "example" {
  name                  = "Test Policy"
  status                = "ACTIVE"
  description           = "This is a test."
  priority              = 1
  groups_included       = [okta_group_Group1.id]
}

data "okta_group" "Group1" {
  name = "Group1"
}


resource "okta_policy_rule_signon" "IdP" {
  name                  = "External IdP"
  policy_id             = okta_policy_signon.example.id
  priority              = 1 
  network_connection    = "ANYWHERE"
  identity_provider     = "SPECIFIC_IDP"
  identity_provider_ids = data.okta_idp_saml.IdP.id
  authtype              = "ANY"
  behaviors             = []
  risc_level            = "ANY"
  access                = "ALLOW"
  session_lifetime      = 0
  session_idle          = 60 
}

data "okta_idp_saml" "Idp" {
    name = "External IdP"
}
```

On provider version 3.25.0 when running the Terraform it creates this rule:
![image](https://user-images.githubusercontent.com/24887992/169620702-f5e116a4-b65c-4129-b07b-9ba99061d974.png)

On version 3.25.1 and later it creates this rule, which has Identity provider set to ANY instead of the IdP provided in the Terraform.
![image](https://user-images.githubusercontent.com/24887992/169620817-0cdb69f8-34d7-4c87-afe1-4dc3908a00ab.png)

**Comment by monde** [comment]
Thanks @kellycampbe I'll look into this.

**Comment by monde** [comment]
The regression seems to be in 0ebeae7c5e062a084809397ca93b194cdef17a94 that fixed https://github.com/okta/terraform-provider-okta/issues/1079

**Comment by monde** [comment]
Released as [v3.28.0](https://github.com/okta/terraform-provider-okta/releases/tag/v3.28.0).
